include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,459 @@
+// Function _getFileTimeDiff()
+// ----------------------------------------------------------------------------
+
+inline double _getFileTimeDiff(std::string const & timestamp_filename)
+{
+    double curr = static_cast<double>(std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now().time_since_epoch()).count());
+    std::ifstream timestamp_file;
+    timestamp_file.open(timestamp_filename.c_str());
+
+    if (timestamp_file.is_open())
+    {
+        std::string str_time;
+        std::getline(timestamp_file, str_time);
+        timestamp_file.close();
+        double d_time;
+        lexicalCast(d_time, str_time);
+        return curr - d_time;
+    }
+
+    return curr;
+}
+
+// ----------------------------------------------------------------------------
+// Function _getNumbersFromString()
+// ----------------------------------------------------------------------------
+
+inline String<int> _getNumbersFromString(std::string const & str)
+{
+    String<int> numbers;
+    StringSet<std::string> set;
+    strSplit(set, str, EqualsChar<'.'>(), false);
+    for (auto & num : set)
+        appendValue(numbers, lexicalCast<int>(num));
+    return numbers;
+}
+
+// ----------------------------------------------------------------------------
+// Function _readVersionString()
+// ----------------------------------------------------------------------------
+
+inline std::string _readVersionString(std::string const & version_file)
+{
+    std::ifstream myfile;
+    myfile.open(version_file.c_str());
+    std::string line;
+    if (myfile.is_open())
+    {
+        std::getline(myfile,line); // get first line which should only contain the version number
+        if (!(std::regex_match(line, std::regex("^[[:digit:]]+\\.[[:digit:]]+\\.[[:digit:]]+$"))))
+        {
+            line.clear();
+        }
+        if (line == VersionControlTags_<>::UNREGISTERED_APP)
+        {
+            std::cerr << "[SEQAN INFO] :: Thank you for using SeqAn!\n"
+                      << "[SEQAN INFO] :: You might want to regsiter you app for support and version check features?!\n"
+                      << "[SEQAN INFO] :: Just send us an email to seqan@team.fu-berlin.de with your app name and version number.\n"
+                      << "[SEQAN INFO] :: If you don't want to recieve this message anymore set --version_check 2\n\n";
+            line.clear();
+        }
+        myfile.close();
+    }
+
+    return line; // line is an empty string on failure
+}
+
+// ----------------------------------------------------------------------------
+// Function _callServer()
+// ----------------------------------------------------------------------------
+
+inline void _callServer(VersionCheck const me, std::promise<bool> prom)
+{
+    // update timestamp
+    std::string timestamp_filename = me._path + "/" + me._name + ".timestamp";
+    std::ofstream timestamp_file(timestamp_filename.c_str());
```

Comment:
Unless there is a very good reason, try to always use brace-initialization (it is possible for both line preceeding this comment!).